core/include/scipp/core/element/util.h

```diff
@@ -62,4 +62,25 @@ constexpr auto variances = overloaded{
     },
     [](const units::Unit &u) { return u * u; }};
 
+constexpr auto is_sorted_common = overloaded{
+    core::element::arg_list<
+        std::tuple<bool, double, double>, std::tuple<bool, float, float>,
+        std::tuple<bool, int64_t, int64_t>, std::tuple<bool, int32_t, int32_t>,
+        std::tuple<bool, std::string, std::string>>,
+    transform_flags::expect_no_variance_arg<1>,
+    [](units::Unit &out, const units::Unit &left, const units::Unit &right) {
+      core::expect::equals(left, right);
+      out = units::dimensionless;
+    }};
+
+constexpr auto is_sorted_nondescending = overloaded{
```

Comment:
Why isn't this just called `is_sorted_ascending`? similar for overloaded below?

Reply:
I chose this name because `ascending` can be interpreted as returning `false` for a sequence like `[1,2,2,3]`. I basically adopted the nomenclature from https://en.cppreference.com/w/cpp/algorithm/is_sorted.